package.json

```diff
@@ -135,6 +137,7 @@
     "prop-types": "^15.7.2",
     "pusher-js": "^7.0.3",
     "rails-erb-loader": "^5.5.2",
+    "react-inlinesvg": "^2.3.0",
```

Comment:
Just as some additional commentary - I looked into using something like this a while ago, but at the time put it on hold because there are so many places we currently create accessible names by using an ID on the title element and aria-labelledby (and none of these libs seem to allow you to pass an ID specifically to the title element).

Since then I've learned that labelling controls by using the title attribute is not great for assistive tech (it's unreliably handled by different screen readers). So I think we can definitely use this, and as we refactor out all the duplicated inline SVG code we can have accessible labels in mind.

Once this is merged we can create a separate issue to cover refactoring/cleaning up current SVG code.

Reply:
@nickytonline might need to tag you in for a look at an issue we're having...

When `react-inlinesvg` is installed, a bunch of tests in `ArticleCoverImage.test.jsx` are failing. It seems to all relate to testing change events, e.g.:

```js
const fakeSuccessMessage = `{ "action": "success", "link": "/some-fake-image.jpg" }`;
fireEvent.change(nativeInput, {
  target: { value: fakeSuccessMessage },
});
```

After which I can see the input has successfully updated with the correct value, but the `onChange` callback on the input never seems to be called 🙃  

`react-inlinesvg` isn't actually used by any of the current app code, and even looking at its dependencies I'm not sure exactly _why_ it's interfering... We can make the test weirdness go away by mocking the lib in jest but it would be good to understand what's going on first.

Maybe you have some time to pair on it this week?

Reply:
Actually, @ludwiczakpawel I am thinking the sensible thing to do would be if you're cool to split off the addition of `react-inlinesvg`/adding the Icon component to a separate issue/PR so we can prioritise/look into the test weirdness more at a later stage?

I don't want your PR getting held up ages by it, but I'm also cautious to spend too much time on it right at this moment. What do you think?

Reply:
@aitchiss without adding this lib, the entire <Icon /> component wouldn't make sense, and therefore several stories for Buttons and CTAs _with icon_ (these component would have to be changed too). So I'd rather wait for it to be solved here, especially this PR isn't really blocking any work right now.

Reply:
That's cool - it just might take a while to get back round to digging into this 🕵️‍♀️

Reply:
Looking at this now.

Reply:
I had a short look at this locally and I was able to get things working fairly quickly adding also `babel-plugin-module-resolver` and updating the `babel.config.js`. It currently isn't very happy with the `@img` alias and I think this [may be a limitation of how paths are resolved](https://github.com/airbnb/babel-plugin-inline-react-svg/pull/17).

In terms of the reusable `<Icon />` component, I think it could work the same, but wouldn't support the possibility to pass a string `src` path that gets dynamically imported at runtime. e.g. something like this should work:

```js
import CogIcon from '../../assets/images/cog.svg'

const MyComponent = () => {
  return <CrayonsIcon Icon={CogIcon}/>
}
```

```jsx
// This is the "Icon" component that basically just enhances the given icon with the correct classnames based on other props
const CrayonsIcon = ({ Icon }) => {
  return <Icon className="custom crayonsy classnames" anyOtherAttrs />
}
```

So basically all the `CrayonsIcon` component would do is apply required styling

Reply:
So... What's your suggestion?